### PR TITLE
Fix for bug #593

### PR DESCRIPTION
**Pull request by tanelihuuskonen**
I fixed bug #593 and added one test case.  My commits look a bit funny as I forgot the commandment, "Thou shalt not amend an already published commit."

I know the licensing policy and agree to it.

### CODE REVIEW

File: core/src/main/java/com/google/zxing/oned/Code128Writer.java [hunk 1]
```diff
@@ -182,19 +176,74 @@ public BitMatrix encode(String contents,
     return result;
   }
 
-  private static boolean isDigits(CharSequence value, int start, int length) {
-    int end = start + length;
+  // Returns 1 for FNC_1, 2 for a pair of digits, -1 for a single digit,
+  // 0 otherwise
+  private static int codeCLength(CharSequence value, int start) {
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/oned/Code128Writer.java]: Is it clearer just to make a private enum to express the possible return types?

File: core/src/main/java/com/google/zxing/oned/Code128Writer.java [hunk 2]
```diff
@@ -182,19 +176,74 @@ public BitMatrix encode(String contents,
...
+    if (c == ESCAPE_FNC_1) {
+      return 1;
+    }
+    if (c < '0' || c > '9') {
+      return 0;
+    }
+    if (start+1 >= last) {
+      return -1;
+    }
+    c = value.charAt(start+1);
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/oned/Code128Writer.java]: Nit: spaces around operator

**Comment by tanelihuuskonen** [comment]
Thanks for comments, did the changes.

File: core/src/test/java/com/google/zxing/oned/Code128WriterTestCase.java [hunk 3]
```diff
@@ -90,6 +108,10 @@ public void testEncodeWithFunc4() throws WriterException {
     assertEquals(expected, actual);
   }
 
+  private static BitArray matrixToArray(BitMatrix result) {
```
Comment (srowen) [inline core/src/test/java/com/google/zxing/oned/Code128WriterTestCase.java]: This is just used one place, and it doesn't really make something into an array, just returns one row right?

File: core/src/test/java/com/google/zxing/oned/Code128WriterTestCase.java [hunk 4]
```diff
@@ -22,8 +22,11 @@
 
 import com.google.zxing.BarcodeFormat;
 import com.google.zxing.Writer;
+import com.google.zxing.Result;
```
Comment (srowen) [inline core/src/test/java/com/google/zxing/oned/Code128WriterTestCase.java]: Super nit but can you alphabetize these?

File: core/src/test/java/com/google/zxing/oned/Code128WriterTestCase.java [hunk 5]
```diff
@@ -32,14 +35,18 @@
...
   public static final String QUIET_SPACE = "00000";
   public static final String STOP = "1100011101011";
 
   private Writer writer;
+  private Code128Reader reader;
 
   @Before
   public void setup() {
     writer = new Code128Writer();
+    reader = new Code128Reader();
```
Comment (srowen) [inline core/src/test/java/com/google/zxing/oned/Code128WriterTestCase.java]: This is mostly a test of encoding, but yeah seems OK if this needs some verification

**Comment by srowen** [comment]
OK minus two more tiny comments, this seems fine

**Comment by tanelihuuskonen** [comment]
Done.

**Pull request merged**